### PR TITLE
Added nimbus-cli documentation

### PR DESCRIPTION
**Pull request by jhugman**
Fixes [EXP-3477](https://mozilla-hub.atlassian.net/browse/EXP-3477).

## Description

For ease of reading, two files are rendered here.

- [Install guide, rendered version](https://github.com/mozilla/experimenter-docs/blob/df337cea7ca9adc3604bf642acff9cb51d715ebc/docs/deep-dives/mobile/nimbus-cli/10-install.mdx).
- [Getting Started, rendered version](https://github.com/mozilla/experimenter-docs/blob/df337cea7ca9adc3604bf642acff9cb51d715ebc/docs/deep-dives/mobile/nimbus-cli/20-getting-started.mdx)

## Issue that this pull request resolves (optional)

This will link to and close an issue in GitHub. Replace `experimenter` with the repository where the issue lives and replace `0000` with the issue number. Remove this section if not applicable.

Closes: mozilla/experimenter#0000

## Other information (optional)

Any other information or requests important to this pull request. Remove this section if not applicable.

If you're not certain that your Markdown will render correctly, you can include this line:
I have not ran the project locally with my changes and it would be be ideal for the reviewer to check into my branch and ensure images etc. are rendering as expected.

### CODE REVIEW

File: docs/deep-dives/mobile/nimbus-cli/configuring-via-environment-var.mdx [hunk 1]
```diff
@@ -0,0 +1,51 @@
+---
+id: nimbus-cli-environment-vars
+title: Configuration via environment variables
+slug: /nimbus-cli/environment-variables
+---
+
+## `adb` and `xcrun`
```
Comment (eliserichards) [inline docs/deep-dives/mobile/nimbus-cli/configuring-via-environment-var.mdx:7]: Can we nest this under "Prerequisites"?

File: docs/deep-dives/mobile/nimbus-cli/configuring-via-environment-var.mdx [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+The app supposes that `adb` and `xcrun` are on the `PATH`. If they are not (or aren't spelled `adb` or `adb.exe`), then you can configure them via the `ADB_PATH` and `XCRUN_PATH` environment variables.
+
+```sh
+ADB_PATH=C:\Program-Tools\bin\adb.exe nimbus-cli
```
Comment (eliserichards) [inline docs/deep-dives/mobile/nimbus-cli/configuring-via-environment-var.mdx:12]: is this supposed to be `export ABD_PATH...`? Also, what is the `nimbus-cli` at the end doing? I don't think I understand this command

File: docs/deep-dives/mobile/nimbus-cli/configuring-via-environment-var.mdx (hunk 1 above)
Comment (eliserichards) [reply docs/deep-dives/mobile/nimbus-cli/configuring-via-environment-var.mdx:7]: Maybe also link to this section in `install.mdx` 🤔

File: docs/deep-dives/mobile/nimbus-cli/install.mdx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+### For working with Android:
+
+- Install [`adb`](https://developer.android.com/tools/adb), with Android Studio, or via [Platform Tools](https://developer.android.com/tools/releases/platform-tools).
+
+### For working with iOS:
+
+- Install Xcode.
+- Run `xcode-select --install`
+
+## Installing `nimbus-cli` on macOS or Linux
```
Comment (eliserichards) [inline docs/deep-dives/mobile/nimbus-cli/install.mdx:22]: Perfect! This is exactly how it looks for me (mac OS)

File: docs/deep-dives/mobile/nimbus-cli/instrumentation.mdx [hunk 4]
```diff
@@ -0,0 +1,34 @@
+---
+id: nimbus-cli-instrumentation
+title: Instrumentation Guide
+slug: /nimbus-cli/instrumentation
+---
+
+The `nimbus-cli` is a tool to communicate with the Nimbus SDK that is embedded in an app.
+
+- [Instrumenting Android apps for testing](/getting-started/engineers/getting-started-for-android-engineers#instrumenting-the-app-for-testing).
```
Comment (eliserichards) [inline docs/deep-dives/mobile/nimbus-cli/instrumentation.mdx:9]:
```suggestion
## Instrumentation for mobile apps

- [Instrumenting Android apps for testing](/getting-started/engineers/getting-started-for-android-engineers#instrumenting-the-app-for-testing).
```

File: docs/deep-dives/mobile/nimbus-cli/instrumentation.mdx [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+- [Instrumenting iOS apps for testing](/getting-started/engineers/getting-started-for-ios-engineers#instrumenting-the-app-for-testing).
+
+[Adding a new app to the cli](https://github.com/mozilla/application-services/blob/main/components/support/nimbus-cli/src/config.rs) is less straightforward, but not hard. You should start collecting the following information:
```
Comment (eliserichards) [inline docs/deep-dives/mobile/nimbus-cli/instrumentation.mdx:12]:
```suggestion
## Adding new applications

[Adding a new app to the cli](https://github.com/mozilla/application-services/blob/main/components/support/nimbus-cli/src/config.rs) is less straightforward, but not hard. You should start collecting the following information:
```

File: docs/deep-dives/mobile/nimbus-cli/opening-and-resetting.mdx [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+- `test-feature`
+- `apply-file`
+- `log-state`
+- `open`.
+
+In addition, the `enroll`, `open` and `test-feature` commands, can launch the app with a deeplink.
+
+# Deeplinking
+
+`nimbus-cli` knows the apps` deeplinking schemes, and if no `://` is detected in the deeplink, then the deeplink scheme is appended.
```
Comment (eliserichards) [inline docs/deep-dives/mobile/nimbus-cli/opening-and-resetting.mdx:20]:
```suggestion
`nimbus-cli` knows the apps deeplinking schemes, and if no `://` is detected in the deeplink, then the deeplink scheme is appended.
```

File: docs/deep-dives/mobile/nimbus-cli/opening-and-resetting.mdx [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+
+For each of the following, a `--reset-app`
```
Comment (eliserichards) [inline docs/deep-dives/mobile/nimbus-cli/opening-and-resetting.mdx:22]: 👀

File: docs/deep-dives/mobile/nimbus-cli/opening-and-resetting.mdx [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+
+opens the deeplink (i.e. not sending directly to the app) with:
+
+```sh
+$ xcrun simctl openurl \
+        booted \
+        fennec://deep-link?url=settings/general
+```
+
+By default, each of the commands that launches the app also kills the app right before; you can send the app a deeplink without killing it, using the `--no-clobber` property.
```
Comment (eliserichards) [inline docs/deep-dives/mobile/nimbus-cli/opening-and-resetting.mdx:55]: ❤️ `--no-clobber` ❤️

File: docs/deep-dives/mobile/nimbus-cli/opening-and-resetting.mdx [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+
+For Android, opening a deeplink uses the same mechanism as launching the app: using Intents. Thus, we can enroll in an experiment and then open the app with a deeplink.
+
+```sh
+nimbus-cli --app fenix --channel developer enroll release-android-onboarding-redesign --branch treatment-a --deeplink onboarding/privacy-first
+```
+
+## `test-feature`
+
+Similarly, you can use a `--deeplink` to enroll a []`test-feature`](/nimbus-cli/test-feature) experiment:
```
Comment (eliserichards) [inline docs/deep-dives/mobile/nimbus-cli/opening-and-resetting.mdx:76]:
```suggestion
Similarly, you can use a `--deeplink` to enroll a [`test-feature`](/nimbus-cli/test-feature) experiment:
```

File: docs/deep-dives/mobile/nimbus-cli/opening-and-resetting.mdx [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+
+# Terminating and Resetting the app
+
+`open`, `enroll` and `test-feature` all support a `--reset-app` option, which serves to reset the app back to its just-installed state before opening.
```
Comment (eliserichards) [inline docs/deep-dives/mobile/nimbus-cli/opening-and-resetting.mdx:80]: Oh I think this is where you're talking about reset app. I guess you can remove the one above where I said " 👀 "

File: docs/deep-dives/mobile/nimbus-cli/working-with-files.mdx [hunk 11]
```diff
@@ -0,0 +1,58 @@
+---
+id: nimbus-cli-working-with-files
+title: Working with Files
+slug: /nimbus-cli/working-with-files
+---
+
+## `fetch`
+
+You can download from a server either the current list of all experiments:
```
Comment (eliserichards) [inline docs/deep-dives/mobile/nimbus-cli/working-with-files.mdx:9]:
```suggestion
You can download from a server the current list of all experiments:
```

File: docs/deep-dives/mobile/nimbus-cli/working-with-files.mdx [hunk 12]
```diff
@@ -0,0 +1,58 @@
...
+
+```sh
+nimbus-cli --app fenix --channel developer fetch archived.json
+```
+
+Use `preview`, `stage` or `stage/preview` to specify a Remote Settings instance.
```
Comment (eliserichards) [inline docs/deep-dives/mobile/nimbus-cli/working-with-files.mdx:15]: What does `stage/preview` do? is that different from specifying just one of the two?

File: docs/deep-dives/mobile/nimbus-cli/working-with-logs.mdx [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+
+Log files are especially helpful to developers, but not always very easy to access.
+
+`nimbus-cli` uses `adb logcat` on Android, and on iOS it uses the Unix commands `find` and `tail`.
+
+These logs are useful when examining the internal state of the app, or attaching to bug reports.
+
+> For iOS the logs that are displayed and captured are the app's logs.
+> For Android, the logs are for the whole device.
+> For Android devices also used for personal usage, you should probably not upload such logs anywhere.
```
Comment (eliserichards) [inline docs/deep-dives/mobile/nimbus-cli/working-with-logs.mdx:15]:
```suggestion
:::info
For iOS the logs that are displayed and captured are the app's logs.
For Android, the logs are for the whole device.
For Android devices also used for personal usage, you should probably not upload such logs anywhere.
:::
```

File: docs/deep-dives/mobile/nimbus-cli/working-with-logs.mdx [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+
+> On iOS, because the `reset-app` command deletes the app contaner, the `tail-logs` command is only available after the first run after a reset, and stops after a reset.
+
+## `log-state`
+
+This command is an analog for the experiments menu in the secret settings of Fenix and Firefox for iOS.
+
+It relaunches the app, then asks the SDK to dump the state of the Nimbus SDK to the logs. For example:
+
+For example, this is the
```
Comment (eliserichards) [inline docs/deep-dives/mobile/nimbus-cli/working-with-logs.mdx:50]: this is the... :)

**Review by eliserichards — COMMENTED** [review]
This looks amazing! Thank you so much for spending the time to write all of this up. I have a few suggestions, but overall I think it is 🔥

File: docs/deep-dives/mobile/nimbus-cli/opening-and-resetting.mdx (hunk 6 above)
Comment (jrbenny35) [reply docs/deep-dives/mobile/nimbus-cli/opening-and-resetting.mdx:20]: I think it needs that apostrophe after ```apps```

File: docs/deep-dives/mobile/nimbus-cli/working-with-files.mdx (hunk 12 above)
Comment (jrbenny35) [reply docs/deep-dives/mobile/nimbus-cli/working-with-files.mdx:15]: There's a preview instance on stage too

**Review by jhugman — COMMENTED** [review]
Thanks @eliserichards , these are really helpful comments!

**Review by eliserichards — APPROVED** [review]
Absolutely gorgeous 🤩 🥇 Thank you so much for writing all of this!!

**Pull request merged**